core/runtime/TRTEngine.cpp

```diff
@@ -30,6 +32,12 @@ TRTEngine::TRTEngine(std::vector<std::string> serialized_info)
           std::string("[] = "),
           util::logging::get_logger().get_reportable_severity(),
           util::logging::get_logger().get_is_colored_output_on()) {
+  TRTORCH_CHECK(
+      serialized_info.size() == ENGINE_IDX + 1, "Program to be deserialized targets an incompatible TRTorch ABI");
```

Comment:
Why serialized_info.size() is being compared against ENGINE_IDX + 1?
Shouldn't this be versioned along with TRTorch?

Reply:
wdym versioned? ENGINE_IDX is the largest index, so it verifies the correct size for the array, if we prepend new data, this will stay true, otherwise having some constant would have the same update issue as updating the index